vscode-extension/src/extension.ts

```diff
@@ -233,22 +223,64 @@ async function createNewAIConfig(
   const fileContentBuffer = await vscode.workspace.fs.readFile(fileContentPath);
   const initialContent = fileContentBuffer.toString();
 
-  const doc = await vscode.workspace.openTextDocument({
-    content: initialContent,
-    language: mode,
+  const workspaceUri = vscode.workspace.workspaceFolders
+    ? vscode.workspace.workspaceFolders[0].uri
+    : null;
+
+  // Find the first available untitled file name to suggest as default
+  let firstAvailableUntitledName: string | null = null;
+  let i = 0;
+  while (firstAvailableUntitledName === null) {
+    const fileName = `untitled${i === 0 ? "" : "-" + i}.aiconfig.${mode}`;
+    const filePath = workspaceUri
+      ? path.join(workspaceUri.path, fileName)
+      : fileName;
+
+    if (fs.existsSync(filePath)) {
+      i++;
+    } else {
+      firstAvailableUntitledName = fileName;
+    }
+  }
+
+  const newConfigName = await vscode.window.showInputBox({
+    prompt: "Enter a name for the new AIConfig file",
+    value: firstAvailableUntitledName,
+    validateInput: (input) => {
+      const nameValidationError = validateConfigName(input, mode);
+      if (nameValidationError) {
+        return nameValidationError;
+      }
+
+      return fs.existsSync(input) ? "File already exists" : null;
+    },
   });
 
-  //const doc = await vscode.workspace.openTextDocument(untitledUri);
-  await vscode.window.showTextDocument(doc, {
-    preview: false,
-    viewColumn: vscode.ViewColumn.One,
+  const newConfigFilePath = workspaceUri
+    ? path.join(workspaceUri.path, newConfigName)
+    : newConfigName;
+  const newConfigUri = vscode.Uri.file(newConfigFilePath).with({
+    scheme: "untitled",
```

Comment:
Why is this needed? Tried looking it up on [VSCode's github](https://github.com/microsoft/vscode-uri) but not super clear what untitled schema does:

<img width="868" alt="Screenshot 2024-02-24 at 17 31 42" src="https://github.com/lastmile-ai/aiconfig/assets/151060367/c6a146fd-998f-4e46-a9cd-4d34c9415739">

Reply:
From https://code.visualstudio.com/api/references/document-selector#:~:text=The%20untitled%20scheme%20is%20for,yet%20been%20saved%20to%20disk.

> The untitled scheme is for new files that have not yet been saved to disk.

Which is technically the case until they save it for the first time